src/js/plugins/plugin.search.view.js

```diff
@@ -0,0 +1,317 @@
+class SearchView {
+  constructor(params) {
+    if (!params.selector) {
+      console.warn('BookReader::Search - SearchView must be passed a valid CSS selector');
+      return;
+    }
+
+    this.br = params.br;
+
+    this.matcher = new RegExp('{{{(.+?)}}}', 'g');
+    this.dom = {
+      searchTray: this.renderSearchTray(params.selector),
+    };
+    this.cacheDOMElements();
+    this.bindEvents();
+
+    if (this.br.options.initialSearchTerm) {
+      this.setQuery(params.query);
+    }
+  }
+
+  cacheDOMElements() {
+    this.dom.results = this.dom.searchTray.querySelector('[data-id="results"]');
+    this.dom.resultsCount = this.dom.searchTray.querySelector('[data-id="results_count"]');
+    this.dom.searchField = this.dom.searchTray.querySelector('[name="query"]');
+    this.dom.searchPending = this.dom.searchTray.querySelector('[data-id="searchPending"]');
+    this.dom.mobileSearch = this.buildMobileDrawer();
+    this.dom.toolbarSearch = this.buildToolbarSearch();
+  }
+
+  toggleSearchTray(bool) {
+    const state = typeof bool !== 'boolean' ? this.dom.searchTray.classList.contains('hidden') : bool;
+    this.dom.searchTray.classList.toggle('hidden', !state);
+  }
+
+  toggleResultsCount(bool) {
+    this.dom.resultsCount.classList.toggle('visible', bool);
+  }
+
+  updateResultsCount(results) {
+    this.dom.resultsCount.innerText = `(${results} result${results != 1 ? 's' : ''})`;
+    this.toggleResultsCount(true);
+  }
+
+  setQuery(query) {
+    this.dom.searchField.value = query;
+  }
+
+  emptyMatches() {
+    this.dom.results.innerHTML = '';
+  }
+
+  removeResultPins() {
+    this.br.$('.BRnavpos .BRsearch').remove();
+  }
+
+  clearSearchFieldAndResults() {
+    this.toggleResultsCount(false);
+    this.removeResultPins();
+    this.emptyMatches();
+    this.setQuery('');
+  }
+
+  renderSearchTray(selector) {
```

Comment:
ah ok, i see why you sent the selector string through instead of the dom node.

Reply:
Yeah, trying to use as little jQuery as possible in new BR code. Still had to use it for the event system, but we can set and trigger arbitrary events on the document without it. Refactoring that would require touching all of BookReader, though.